integration/proxy/teleterm_test.go

```diff
@@ -196,13 +195,24 @@ func newMockTSHDEventsServiceServer(t *testing.T, tc *libclient.TeleportClient,
 
 	grpcServer := grpc.NewServer()
 	api.RegisterTshdEventsServiceServer(grpcServer, tshdEventsService)
-	t.Cleanup(grpcServer.GracefulStop)
 
+	serveErr := make(chan error)
 	go func() {
-		err := grpcServer.Serve(ls)
-		assert.NoError(t, err)
+		serveErr <- grpcServer.Serve(ls)
 	}()
 
+	t.Cleanup(func() {
+		grpcServer.GracefulStop()
+
+		// For test cases that did not send any grpc calls, test may finish
+		// before grpcServer.Serve is called and grpcServer.Serve will return
+		// grpc.ErrServerStopped.
+		err := <-serveErr
+		if len(tshdEventsService.callCounts) == 0 || err != grpc.ErrServerStopped {
+			require.NoError(t, err)
+		}
```

Comment:
I don't follow why the counts are relevant here, could this be:

```suggestion
		if err != grpc.ErrServerStopped {
			assert.NoError(t, err, "gRPC server exited with error")
		}
```

(I suggest the require->assert change nonetheless.)

Reply:
Yeah that's probably fine too